internal/provider/config_resource.go

```diff
@@ -99,6 +110,28 @@ func (m *MTEConfigResource) Configure(ctx context.Context, req resource.Configur
 	m.client = resourceData.client
 }
 
+func (m *MTEConfigResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
```

Comment:
What does this do in the flow? Is this not done automatically via the schema?

Reply:
This one is used for when there are two optional fields where one of them needs to be present (so an either/or scenario), you can't just do that through the optional or required flag, additional logic is needed